### PR TITLE
Clarify IntegratedGradients target semantics for binary and multilabel settings

### PR DESCRIPTION
**Pull request by ParthAgarwalCode**
This PR clarifies IntegratedGradients target semantics across prediction modes.

Changes:
- add `decision_threshold` for binary/multilabel default target inference
- use `decision_threshold` instead of hardcoded `0.5`
- document binary default behavior
- document multilabel default behavior
- clarify that `target_class_idx` in binary behaves like target label selection
- clarify that multilabel default attribution uses the sum of selected logits

This does not change the IG algorithm itself, and preserves existing default behavior when `decision_threshold=0.5`.

### CODE REVIEW

File: pyhealth/interpret/methods/integrated_gradients.py [hunk 1]
```diff
@@ -193,6 +195,7 @@ def __init__(self, model: BaseModel, use_embeddings: bool = True, steps: int = 5
 
         self.use_embeddings = use_embeddings
         self.steps = steps
+        self.decision_threshold = decision_threshold
```
Comment (Logiquo) [inline pyhealth/interpret/methods/integrated_gradients.py:199]: I dont see this field used anywhere.

**Comment by Logiquo** [comment]
This issue has been fix by https://github.com/sunlabuiuc/PyHealth/pull/926 already.
